extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/ClassNames.java

```diff
@@ -465,4 +465,22 @@ private static DotName createConstant(String fqcn) {
             "io.quarkus.hibernate.orm.deployment.HibernateUserTypeProcessor");
 
     public static final DotName GRAAL_VM_FEATURES = createConstant("io.quarkus.hibernate.orm.deployment.GraalVMFeatures");
+
+    public static final List<DotName> SERVICE_PROVIDERS = List.of(
+            // Accessed in org.hibernate.query.sqm.internal.SqmCriteriaNodeBuilder.<init>
+            createConstant("org.hibernate.query.criteria.spi.CriteriaBuilderExtension"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.customized.QuarkusIntegratorServiceImpl.<init>
+            createConstant("org.hibernate.integrator.spi.Integrator"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.customized.QuarkusStrategySelectorBuilder.buildSelector
+            createConstant("org.hibernate.boot.registry.selector.StrategyRegistrationProvider"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.recording.RecordableBootstrap.applyServiceContributors
+            createConstant("org.hibernate.service.spi.ServiceContributor"),
+            // Accessed in org.hibernate.internal.FastSessionServices.<init>
+            createConstant("org.hibernate.event.spi.EventManager"),
+            // Accessed in org.hibernate.query.internal.QueryEngineImpl.sortedFunctionContributors
+            createConstant("org.hibernate.boot.model.FunctionContributor"),
+            // Accessed in org.hibernate.event.spi.EventEngine.<init>
+            createConstant("org.hibernate.event.spi.EventEngineContributor"),
+            // Accessed in org.hibernate.service.internal.SessionFactoryServiceRegistryFactoryImpl.buildServiceRegistry
+            createConstant("org.hibernate.service.spi.SessionFactoryServiceContributor"));
```

Comment:
If we do this, we can probably remove that:

https://github.com/quarkusio/quarkus/blob/4e934d084e72f1b73c0ce85b0469f3a49f5ae1bc/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java#L120-L125